test/e2e.method.send.js

```diff
@@ -0,0 +1,199 @@
+
+        it('errors on revert', async function(){
+            try {
+                await instance
+                    .methods
+                    .reverts()
+                    .send({from: accounts[0]});
+
+                assert.fail();
+
+            } catch(err){
+                var receipt = utils.extractReceipt(err.message);
+
+                assert(err.message.includes('revert'))
+                assert(receipt.status === false);
+            }
+        });
+    });
+
+    describe('ws', function() {
+        // Websockets extremely erratic for geth instamine...
+        if (process.env.GETH_INSTAMINE) return;
+
+        before(async function(){
+            // Ganache runs ws and http over the same port
+            var port = process.env.GANACHE ?  8545 : 8546;
+
+            web3 = new Web3('ws://localhost:' + port);
+            accounts = await web3.eth.getAccounts();
+
+            basic = new web3.eth.Contract(Basic.abi, basicOptions);
+            instance = await basic.deploy().send({from: accounts[0]});
+        })
+
+        it('returns a receipt', async function(){
+            var receipt = await instance
+                .methods
+                .setValue('1')
+                .send({from: accounts[0]});
+
+            assert(receipt.status === true);
+            assert(web3.utils.isHexStrict(receipt.transactionHash));
+        });
+
+        it('errors on OOG', async function(){
+            try {
+                await instance
+                    .methods
+                    .setValue('1')
+                    .send({from: accounts[0], gas: 100});
+
+                assert.fail();
+
+            } catch(err){
+                assert(err.message.includes('gas'))
+            }
+        });
+
+        it('errors on revert', async function(){
+            try {
+                await instance
+                    .methods
+                    .reverts()
+                    .send({from: accounts[0]});
+
+                assert.fail();
+
+            } catch(err){
+                var receipt = utils.extractReceipt(err.message);
+
+                assert(err.message.includes('revert'))
+                assert(receipt.status === false);
+            }
+        });
+
+        it('fires the transaction event', function(done){
+            instance
+                .methods
+                .setValue('1')
+                .send({from: accounts[0]})
+
+                .on('transactionHash', hash => {
+                    assert(web3.utils.isHex(hash))
+                    done();
+                })
+        });
+
+        it('fires the receipt event', function(done){
+            instance
+                .methods
+                .setValue('1')
+                .send({from: accounts[0]})
+
+                .on('receipt', receipt => {
+                    assert(receipt.status === true)
+                    done();
+                })
+        })
+
+        it('fires the confirmation handler', function(){
```

Comment:
why not a top-level ``async``? ``it('fires the confirmation handler', async function () {``

Reply:
Normally would use `done` here because we need to wait for the EventEmitter handler to fire. However, I want to make async calls to fetch the blockNumber, and mocha complains about await and done being used in the same function. For example:
```js
it('a test, async function(done){
  await somePromise();
  done();
})
```
throws...
```
Error: Resolution method is overspecified. Specify a callback *or* return a Promise; not both.
```